src/thor/triplegbuilder.cc

```diff
@@ -481,6 +481,45 @@ void AddTransitNodes(TripLeg_Node* trip_node,
   }
 }
 
+void SetTripEdgeRoadClass(TripLeg_Edge* trip_edge,
+                          const DirectedEdge* directededge,
+                          const GraphTile* graphtile,
+                          GraphReader& graphreader) {
+  trip_edge->set_road_class(GetTripLegRoadClass(directededge->classification()));
+  // If this is a ramp it may have been reclassified in graph enhancer.
+  // To restore the original road class for motorway_links, we check if any of the adjacent edges is
+  // a motorway.
+  if (directededge->use() == Use::kRamp) {
+    const DirectedEdge* opposing_edge = graphreader.GetOpposingEdge(directededge, graphtile);
+    for (const auto& edge : {directededge, opposing_edge}) {
+      GraphId endnode = edge->endnode();
+      const GraphTile* tile = edge->leaves_tile() ? graphreader.GetGraphTile(endnode) : graphtile;
```

Comment:
no need to do this, you can use the other signature of GetGraphTile and pass it the tile you have so it cna make use of it without you checking if it leaves the tile etc